src/custom/pages/Swap/SwapMod.tsx

```diff
@@ -258,6 +246,25 @@ export default function Swap({ history, location, className }: RouteComponentPro
 
   const swapCallbackError = swapFlowContext ? null : 'Missing dependencies'
 
+  // handle swap when native token is detected as sell token
+  const handleNativeWrapAndSwap = () => {
+    if (!chainId) throw new Error('Need to be connected')
+
+    // switch to wrapped native currency
+    onCurrencySelection(Field.INPUT, WRAPPED_NATIVE_CURRENCY[chainId])
+    // set swap state
+    openSwapConfirmModal(trade!)
```

Comment:
I'm not a huge fan of the non-null assertion operator (!) - why not add a conditional check beforehand just to be safe?

Reply:
Thanks! Already fixed in the next PR's